pkg/storage/ingest/reader.go

```diff
@@ -169,11 +170,28 @@ func (r *PartitionReader) consumeFetches(ctx context.Context, fetches kgo.Fetche
 		})
 	})
 
-	err := r.consumer.consume(ctx, records)
-	if err != nil {
-		level.Error(r.logger).Log("msg", "encountered error processing records; skipping", "min_offset", minOffset, "max_offset", maxOffset, "err", err)
-		// TODO abort ingesting & back off if it's a server error, ignore error if it's a client error
+	boff := backoff.New(ctx, backoff.Config{
+		MinBackoff: 250 * time.Millisecond,
+		MaxBackoff: 2 * time.Second,
+		MaxRetries: 0, // retry forever
+	})
+
+	for boff.Ongoing() {
+		err := r.consumer.consume(ctx, records)
+		if err != nil {
+			level.Error(r.logger).Log(
```

Comment:
We need a metric here, to be able to alert on it (we need to compute the failure rate).

Reply:
Generally speaking, this `for` loop could be an infinite loop in case of a persistent error, so I would like to better understand how we're going to alert on it.

Reply:
I was thinking that we alert on end-to-end latency. If the consume errors are infrequent enough we can probably get away with them without user-visible effects. Since we block consumption the failure rate will either be 0% or 100%, which doesn't sound very insightful.